### DISCUSSION TITLE
Fix image scaling after monitor resolution/rotation has changed.

### DISCUSSION THREAD

**Discussion by ghost** [nix-community/stylix#632] [Q&A]
Hey All,
With my setup I have to use xrandr to arrange my monitors at startup and stylix doesn't take this into account.
The wallpaper is setup but then xrandr runs which rotates my second monitor and then the wallpaper no longer fits the screen. Rebuilding my system doesn't reapply the wallpaper.
Can this be resolved?

**Comment by trueNAHO** [comment]
> With my setup I have to use xrandr to arrange my monitors at startup and stylix doesn't take this into account. The wallpaper is setup but then xrandr runs which rotates my second monitor and then the wallpaper no longer fits the screen. Rebuilding my system doesn't reapply the wallpaper. Can this be resolved?

AFAIK, Stylix currently merely passes `stylix.image` to supported applications as-is without further modification.

There have been some discussions on implementing the wallpaper drawing in Stylix to give us more control:

> 6. Implement slideshow [1] [6] [7] which will arguably be the coolest Stylix feature
>
> -- https://github.com/danth/stylix/issues/534

Although this would resolve your issue, it will take a while for this to be implemented. For the time being, you could work around this limitation by redrawing the wallpaper in your setup after modifying your monitors.

[1]: https://github.com/danth/stylix/issues/63
[6]: https://github.com/danth/stylix/pull/477#issuecomment-2254170449
[7]: https://github.com/danth/stylix/pull/477#issuecomment-2304904560

**Reply by ghost** [reply]
@trueNAHO 
Thanks for the heads up. I'll continue manually redrawing my wallpapers with nitrogen until this feature comes out.

**Comment by danth** [comment]
Which window manager / desktop environment are you using?

**Reply by ghost** [reply]
Hello danth,
I found this issue on XMonad(what I am currently using), Awesome and Herbstluftwm. I haven't tested anymore X11 window managers than those but it would appear that this is a fairly universal X11 issue based on @trueNAHO comment above.

**Reply by danth** [reply]
Stylix is using [feh](https://feh.finalrewind.org/) for most X11 targets. We could potentially replace this with a different wallpaper program which responds to changes in XRandR, if such a program exists.

**Reply by ghost** [reply]
I assumed something like that, unfortunately I don't think any program does exist which can listen for Xrandr changes and update the wallpaper.
If that really is the case I guess there is not a lot we can do about this issue.
